### PR TITLE
fix: Crash in KitConfiguration due to NumberFormatException

### PR DESCRIPTION
**Pull request by Mansi-mParticle**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - Fix rare crash in KitConfiguration caused by NumberFormatException. Updated code to handle NumberFormatException and null cases.

 ## Testing Plan
 - [x] Was this tested locally? If not, explain why.
 - Tested with the sample application and could not reproduce the issue initially. However, after adding a unit test, I was able to reproduce the issue and address it.

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-6548

### CODE REVIEW

File: android-kit-base/src/test/kotlin/com/mparticle/kits/KitConfigurationTest.kt [hunk 1]
```diff
@@ -676,6 +678,97 @@ class KitConfigurationTest {
...
+        """.trimIndent()
+        val jsonConfiguration = JSONObject(jsonData)
+        val method: Method = MockKitConfiguration::class.java.getDeclaredMethod("convertToSparseArray", JSONObject::class.java)
+        method.isAccessible = true
+        val result = method.invoke(kitConfiguration, jsonConfiguration) as SparseBooleanArray
+        Assert.assertEquals(12, result.size())
+    }
+
+    @Test
+    fun testConvertToSparseArray_When_JSON_OBJECT_IS_NULL() {
```
Comment (rmi22186) [inline android-kit-base/src/test/kotlin/com/mparticle/kits/KitConfigurationTest.kt:708]: Isn't this JSON object just an empty object as opposed to a `null`?  Should there be a test case for 
`{"ec" : null}` or no?

Comment (Mansi-mParticle) [reply android-kit-base/src/test/kotlin/com/mparticle/kits/KitConfigurationTest.kt:708]: Add unit test case for null validation

**Review by rmi22186 — APPROVED** [review]

**Review by einsteinx2 — APPROVED** [review]
LGTM, reviewed live on Zoom
